unit_test/src/lexer_cases.c

```diff
@@ -6,95 +6,187 @@
 extern t_exitstatus zero;
 extern t_exitstatus max;
 
+void redirect_all_std(void)
+{
+    cr_redirect_stdout();
+    cr_redirect_stderr();
+}
+TestSuite(lexer, .init=redirect_all_std);
+t_list	*lexer(const char *command_line, t_exitstatus *exit_status);
+
 /*******************************************************************************/
-/*                           Lexer                                             */
+/*                           Lexer_one                                         */
 /*******************************************************************************/
 
-t_list	*lexer(const char *command_line, t_exitstatus *exit_status);
-
-void assert_lexer(char *command_line, char **expected)
+void assert_lexer_one(char *command_line, char **expected)
 {
     t_list *tokens;
-    t_list *head;
 
     tokens = lexer(command_line, &zero);
-    head = tokens;
-    while (tokens)
-    {
-      cr_assert_not_null(expected);
-      cr_expect_str_eq(tokens->content, *expected++);
-      tokens = tokens->next;
-    }
-    ft_lstclear(&head, free);
+    cr_expect_str_eq(tokens->content, *expected);
+    cr_expect(tokens->next == NULL);
+    ft_lstclear(&tokens, free);
 }
 
-Test(lexer, basic)
+Test(lexer, one_ls)
 {
     char *expected[] = {"ls", NULL};
-    assert_lexer("ls", expected);
+    assert_lexer_one("ls", expected);
 }
```

Comment:
I think it is a good idea to also test for an empty string as input: ""

Reply:
Usually one should consider the following for strings:

1. NULL
2. Empty
3. single char/words
5. two chars/words